dlp/src/main/java/com/example/dlp/RiskAnalysis.java

```diff
@@ -169,76 +214,119 @@ private static void calculateCategoricalStats(
 
-      AnalyzeDataSourceRiskRequest request =
-          AnalyzeDataSourceRiskRequest.newBuilder()
-              .setPrivacyMetric(privacyMetric)
+      ProjectTopicName topicName = ProjectTopicName.of(projectId, topicId);
+
+      PublishToPubSub publishToPubSub = PublishToPubSub.newBuilder()
+          .setTopic(topicName.toString())
+          .build();
+
+      // create action to publish job status notifications over Google Cloud Pub/Sub
+      Action action = Action.newBuilder().setPubSub(publishToPubSub).build();
+
+      RiskAnalysisJobConfig riskAnalysisJobConfig =
+          RiskAnalysisJobConfig.newBuilder()
               .setSourceTable(bigQueryTable)
+              .setPrivacyMetric(privacyMetric)
+              .addActions(action)
               .build();
 
-      // asynchronously submit a risk analysis operation
-      OperationFuture<RiskAnalysisOperationResult, RiskAnalysisOperationMetadata>
-          responseFuture = dlpServiceClient.analyzeDataSourceRiskAsync(request);
-
-      // ...
-      // block on response
-      RiskAnalysisOperationResult response = responseFuture.get();
-      CategoricalStatsHistogramBucket results =
-          response.getCategoricalStatsResult().getValueFrequencyHistogramBuckets(0);
-
-      System.out.println(
-          "Most common value occurs " + results.getValueFrequencyUpperBound() + " time(s)");
-      System.out.println(
-          "Least common value occurs " + results.getValueFrequencyLowerBound() + " time(s)");
-
-      for (ValueFrequency valueFrequency : results.getBucketValuesList()) {
-        System.out.println("Value "
-            + valueFrequency.getValue().toString()
-            + " occurs "
-            + valueFrequency.getCount()
-            + " time(s)."
-        );
+      CreateDlpJobRequest createDlpJobRequest =
+          CreateDlpJobRequest.newBuilder()
+              .setParent(ProjectName.of(projectId).toString())
+              .setRiskJob(riskAnalysisJobConfig)
+              .build();
+
+      DlpJob dlpJob = dlpServiceClient.createDlpJob(createDlpJobRequest);
+      String dlpJobName = dlpJob.getName();
+
+      final SettableApiFuture<Boolean> done = SettableApiFuture.create();
+
+      // setup a Pub/Sub subscriber to listen on the job completion status
+      Subscriber subscriber =
+          Subscriber.newBuilder(
+              ProjectSubscriptionName.newBuilder()
+                  .setProject(projectId)
+                  .setSubscription(subscriptionId)
+                  .build(),
+              (pubsubMessage, ackReplyConsumer) -> {
+                if (pubsubMessage.getAttributesCount() > 0
+                    && pubsubMessage.getAttributesMap().get("DlpJobName").equals(dlpJobName)) {
+                  // notify job completion
+                  done.set(true);
+                  ackReplyConsumer.ack();
+                }
+              })
+              .build();
+      subscriber.startAsync();
+
+      // wait for job completion
```

Comment:
Why do you show them this without commenting on what you might do in a production system?

Reply:
Replaced with the following:
```
// Wait for job completion synchronously
// For longer jobs, consider using an asynchronous execution model such as Cloud Functions
```

Reply:
Done.